interpreter/src/main/scala/pl/touk/nussknacker/engine/types/EspTypeUtils.scala

```diff
@@ -113,17 +120,55 @@ object EspTypeUtils {
     }
   }
 
+  private def getAnnotationOption[T <: Annotation](method: Method, annotationClass: Class[T]): Option[T] =
+    if (method.isAnnotationPresent(annotationClass)) Some(method.getAnnotation(annotationClass)) else None
+
   private def extractMethod(method: Method)
-    = MethodInfo(extractParameters(method), extractMethodReturnType(method), extractNussknackerDocs(method), method.isVarArgs)
+                           (implicit settings: ClassExtractionSettings): List[MethodInfo] =
+    getAnnotationOption(method, classOf[GenericType]) match {
+      case Some(value) => extractGenericMethod(method, value)
+      case None => extractRegularMethod(method)
+    }
+
+  private def extractGenericMethod(method: Method, genericType: GenericType)
+                                  (implicit settings: ClassExtractionSettings): List[MethodInfo] = {
+    val typeFunctionClass = genericType.typingFunction()
+    val typeFunctionConstructor = typeFunctionClass.getDeclaredConstructor()
+    val typeFunctionInstance = typeFunctionConstructor.newInstance()
```

Comment:
let's put try/catch around three lines above (maybe in extracted method) and thow nicer exception?